src/Uno.UWP/System/Power/PowerManager.cs

```diff
@@ -0,0 +1,112 @@
+﻿#if __MOBILE__
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Windows.System.Power
+{
+	public partial class PowerManager
+	{
+		private static EventHandler<object> _powerSupplyStatusChanged;
+		private static EventHandler<object> _energySaverStatusChanged;
+		private static EventHandler<object> _remainingChargePercentChanged;
+		private static EventHandler<object> _batteryStatusChanged;
+
+		public static BatteryStatus BatteryStatus => GetBatteryStatus();
+
+		public static EnergySaverStatus EnergySaverStatus => GetEnergySaverStatus();
+
+		public static PowerSupplyStatus PowerSupplyStatus => GetPowerSupplyStatus();
+
+		public static int RemainingChargePercent => GetRemainingChargePercent();
+
+		public static event EventHandler<object> PowerSupplyStatusChanged
+		{
+			add
+			{
+				if (_powerSupplyStatusChanged == null)
+				{
+					StartPowerSupplyStatusMonitoring();
+				}
+				_powerSupplyStatusChanged += value;
+			}
+			remove
+			{
+				_powerSupplyStatusChanged -= value;
+				if (_powerSupplyStatusChanged == null)
+				{
+					EndPowerSupplyStatusMonitoring();
+				}
+			}
+		}
+
+		public static event EventHandler<object> EnergySaverStatusChanged
+		{
+			add
+			{
+				if (_energySaverStatusChanged == null)
+				{
+					StartEnergySaverStatusMonitoring();
+				}
+				_energySaverStatusChanged += value;
+			}
+			remove
+			{
+				_energySaverStatusChanged -= value;
+				if (_energySaverStatusChanged == null)
+				{
+					EndEnergySaverStatusMonitoring();
+				}
+			}
+		}
+
+		public static event EventHandler<object> RemainingChargePercentChanged
+		{
+			add
+			{
+				if (_remainingChargePercentChanged == null)
+				{
+					StartRemainingChargePercentMonitoring();
+				}
+				_remainingChargePercentChanged += value;
+			}
+			remove
+			{
+				_remainingChargePercentChanged -= value;
+				if (_remainingChargePercentChanged == null)
+				{
+					EndRemainingChargePercentMonitoring();
+				}
+			}
+		}
+
+		public static event EventHandler<object> BatteryStatusChanged
+		{
+			add
+			{
+				if (_batteryStatusChanged == null)
+				{
+					StartBatteryStatusMonitoring();
+				}
+				_batteryStatusChanged += value;
+			}
+			remove
+			{
+				_batteryStatusChanged -= value;
+				if (_batteryStatusChanged == null)
+				{
+					EndBatteryStatusMonitoring();
+				}
+			}
+		}
+
+		internal static void RaiseRemainingChargePercentChanged() => _remainingChargePercentChanged?.Invoke(null, null);
```

Comment:
UWP returns null for args as well as sender?

Reply:
Will check :-)

Reply:
![image](https://user-images.githubusercontent.com/1075116/58122745-9acaa600-7c0a-11e9-9fd8-03428168df0e.png)

Seems that's the case for these events. `sender` is `null` as the class is static, and the second parameter is `null` as well :-D .